### DISCUSSION TITLE
with prisma how do i update the UI instantly while doing an async call to the database?

### DISCUSSION THREAD

**Discussion by diontron** [prisma/prisma#11666] [Q&A] [answered]
im a little new to modern JS

i come from PHP, and one problem with livewire/hotwire type stuff is that we need to wait for backend before UI changes are reflected.

i'm wondering if its possible for me to have UI update instantly while doing async call to database in background with Prisma

so for example, let's say i have a todo list. when a new item is created, i want the `todos` object to change in UI immediately, while syncing with the `todos` in database in background. both client object and the database table it represents should stay in sync

how do i do this with prisma? or does it do this by default? sorry for noob question

**Comment by psibean** [comment, marked as answer]
This isn't really a prisma question? It depends on the UI framework you're using?

Typically, you make your request, which is an asynchronous call to your API.
Your backend processes the response and returns the result, if your response data contains the newly created data, you then update your frontend state, which would then cause it to re-render (i.e. react/vue/angular).

Ideally you shouldn't be updating your UI before the server / backend has finished processing the request, because what happens if the backend fails to make the creation for some reason? Then suddenly you've updated your UI, received a response with an error, and then you have to clean up the UI because it wasn't actually created.

What you can do, is simply change the state of your frontend component to a loading / processing state (some spinner indicator, disabled buttons, etc), and once the server has responded, update it again (you do this in the then, or after your awaited call depending on which syntax you use).

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
